[v2] Refactor: Define all mockery in .mockery.yaml

majiayu000 opened the pull request:
## Summary
This PR fixes #6815

Refactor to define all mockery configuration in `.mockery.yaml` instead of using `//go:generate mockery` comments in individual files.

This PR modify the way we generate the mocks following the best practice described in: https://vektra.github.io/mockery/v2.53/configuration

- Define packages for mocking in `.mockery.yaml`
- Use `EXPECT()` rather than `On()` (in docs: Use `with-expecter: True` to generate `EXPECT()` methods for your mocks. This is the preferred way to set up your mocks.) 

## Changes
- Updated `.mockery.yaml` with configurations for:
  - flyteplugins packages (pluginmachinery/core, workqueue, k8s, catalog, io)
  - flytestdlib packages (cache, utils, storage, fastcheck, random)
  - runs/repository/interfaces (already had config, removed duplicate directive)
- Removed all `//go:generate mockery` comments from interface files

This change consolidates all mock generation configuration into a single location, making it easier to maintain and ensuring consistency across the codebase.

**File changes**: 13 files changed, 64 insertions(+), 15 deletions(-)

Related to: https://github.com/flyteorg/flyte/pull/6813#discussion_r2649208841

machichima commented on .mockery.yaml:
Could you please add

```yaml
filename: "{{.InterfaceName | snakecase}}.go"
inpackage: false
outpkg: "mocks"
```

machichima reviewed (commented):
Seems like there's some format inconsistency between old and new mocks. We want:
- mock files to be in `mocks/`, with `package mocks`
- mock file name in snake case

Could you please update `.mockery.yaml` and run `make gen` to re-generate mocks files? Thank you!

majiayu000 commented:
@machichima Thanks for the review! I ran `make gen` and pushed the regenerated files. I also updated tests/mocks to match the new mockery output and removed duplicate legacy mocks. I ran `go test ./flyteplugins/... ./flytestdlib/... ./
runs/...` locally; failures were in `flytestdlib/config/tests` and `flytestdlib/otelutils` and seem unrelated to this
change.

machichima commented:
Thanks! Overall LGTM. 
We might need to wait for #6792 to be merged first, then apply some minor changes here. 
Would you mind if I push directly to your branch after that, or would you prefer to handle it yourself?

machichima reviewed (approved):
LGTM. Thank you!

The only place with `On()` left is in `queue/service/queue_service_test.go`. Will need a few refactor there. I think we can more it to follow-up PR

```sh
❯ find ./ -name "*_test.go" -type f -exec grep -l "\.On(" {} \;
./queue/service/queue_service_test.go
```

EngHabu reviewed (approved):
Thank you 🙏🏽

popojk reviewed (commented):
@majiayu000 The PR looks good, thanks for your effort. Can you paste the screenshot to prove that every unit tests runs successfully after refactoring? Then we are good to go, thank you!
